modules/gdscript/gdscript_parser.cpp

```diff
@@ -5947,11 +5947,8 @@ bool GDScriptParser::_is_type_compatible(const DataType &p_container, const Data
 		if (p_expression.kind == DataType::BUILTIN && p_expression.builtin_type == Variant::NIL) {
 			return true;
 		}
-		if (p_expression.kind == DataType::BUILTIN) {
-			return false;
-		}
 		// If it's not a built-in, must be an object
-		return true;
+		return !(p_expression.kind == DataType::BUILTIN);
```

Comment:
Won't this be more readable as:
```cpp
if (p_expression.kind == DataType::BUILTIN) {
	return p_expression.builtin_type == Variant::NIL;
} else {
	return true;
}
```
Or even:
```cpp
return p_expression.kind != DataType::BUILTIN || p_expression.builtin_type == Variant::NIL;
// If it is not a bulit-in type (and therefore an object) or is null
```

Also, `!(x == y)` should probably be `x != y` anyway.